### ISSUE TITLE
Not sure env vars are a good idea

### ISSUE THREAD

**Issue by jgarvin** [pureconfig/pureconfig#8]
I've worked in a bunch of shops where the users are actually Linux users who pick up bits of using the terminal and shell, and use programs created by advanced programmers, but who are not advanced programmers themselves. These users overwhelmingly tend to forget when they have env vars exported in a terminal, and then come asking for support because they think some program is behaving incorrectly because they don't realize the env var is still in effect. They especially get confused by apps that can have a setting set by both env var and config file, and pure would introduce yet another possibility -- config files that do different things or even import different things based on env vars. Imagine the confusion of an env var that affects an app, is used in the config, and where the var may or may not override the config value. Also this means that changing the user running the app could cause arbitrarily different results with the same config file.

**Comment by Phlosioneer** [comment]
This feature probably belongs under the "Let the application handle it" category - if the app needs the home directory variable available in the config file, then the app should provide that variable itself, and let the parser handle it. Tying the parser to the shell, when it's otherwise a totally stand-alone component, is awkward.

**Comment by pureconfig** [comment]
I'm on the fence, but we definitely need a parser flag indicating whether or not env vars can be expanded.

**Comment by dansouza** [comment]
Yeah not everything may come from environment variables, someone might want to use Pure config files to configure the behavior of a web application business process, for example. So I think we should just specify that using external variables is possible by calling `$NAME` and/or `${NAME}` and that those can come from anywhere. So it's up to each parser implementation to provide ways for the developer to specify their own custom mappings to those variables - with helpers that automatically expand environment variables for ease of use.

**Comment by pureconfig** [comment]
@dansouza Makes sense to me

**Comment by shelbyd** [comment]
This seems like a 1.x feature to me.
